libs/deal-ticket/src/components/deal-ticket/deal-ticket-stop-order.tsx

```diff
@@ -0,0 +1,600 @@
+import { timeInForceLabel, useOrder } from '@vegaprotocol/orders';
+import {
+  NoWalletWarning,
+  REDUCE_ONLY_TOOLTIP,
+  useNotionalSize,
+} from './deal-ticket';
+import { TypeToggle } from './type-selector';
+import {
+  useStopOrderFormValues,
+  type StopOrderFormValues,
+} from '../../hooks/use-stop-order-form-values';
+import {
+  DealTicketType,
+  useDealTicketTypeStore,
+} from '../../hooks/use-type-store';
+import { mapFormValuesToStopOrdersSubmission } from '../../utils/map-form-values-to-stop-order-submission';
+import { DealTicketButton } from './deal-ticket-button';
+import { DealTicketFeeDetails } from './deal-ticket-fee-details';
+import { validateExpiration } from '../../utils';
+
+export interface StopOrderProps {
+  market: Market;
+  marketPrice?: string | null;
+  submit: (order: StopOrdersSubmission) => void;
+}
+
+const defaultValues: Partial<StopOrderFormValues> = {
+  type: Schema.OrderType.TYPE_LIMIT,
+  side: Schema.Side.SIDE_BUY,
+  timeInForce: Schema.OrderTimeInForce.TIME_IN_FORCE_FOK,
+  triggerType: 'price',
+  triggerDirection:
+    Schema.StopOrderTriggerDirection.TRIGGER_DIRECTION_RISES_ABOVE,
+  expiryStrategy: Schema.StopOrderExpiryStrategy.EXPIRY_STRATEGY_SUBMIT,
+  size: '0',
+};
+
+const stopSubmit: FormEventHandler = (e) => e.preventDefault();
+
+export const StopOrder = ({ market, marketPrice, submit }: StopOrderProps) => {
+  const { pubKey, isReadOnly } = useVegaWallet();
+  const setDealTicketType = useDealTicketTypeStore((state) => state.set);
+  const [, updateOrder] = useOrder(market.id);
+  const updateStoredFormValues = useStopOrderFormValues(
+    (state) => state.update
+  );
+  const storedFormValues = useStopOrderFormValues(
+    (state) => state.formValues[market.id]
+  );
+  const { handleSubmit, setValue, watch, control, formState } =
+    useForm<StopOrderFormValues>({
+      defaultValues: { ...defaultValues, ...storedFormValues },
+    });
+  const { errors } = formState;
+  const lastSubmitTime = useRef(0);
+  const onSubmit = useCallback(
+    (data: StopOrderFormValues) => {
+      const now = new Date().getTime();
+      if (lastSubmitTime.current && now - lastSubmitTime.current < 1000) {
+        return;
+      }
+      submit(
+        mapFormValuesToStopOrdersSubmission(
+          data,
+          market.id,
+          market.decimalPlaces,
+          market.positionDecimalPlaces
+        )
+      );
+      lastSubmitTime.current = now;
+    },
+    [market.id, market.decimalPlaces, market.positionDecimalPlaces, submit]
+  );
+  const side = watch('side');
+  const expire = watch('expire');
+  const triggerType = watch('triggerType');
+  const triggerPrice = watch('triggerPrice');
+  const timeInForce = watch('timeInForce');
+  const type = watch('type');
+  const rawPrice = watch('price');
+  const rawSize = watch('size');
+
+  if (storedFormValues?.size && rawSize !== storedFormValues?.size) {
+    setValue('size', storedFormValues.size);
```

Comment:
Its a bit of a code smell that we are setting state directly in the body of the function component

I think you could avoid this if we make the fields fully controlled by the zustand store _only_, and then update react-hook-form when they change (like how the default deal ticket works currently)

That said, I wonder if not using react-hook-form is the way to go.. seeing as we are fully controlling every field 🤔

Reply:
IMHO going using zustand store as source only makes things to complicated I prefere to write to store as side effect of change in form.

Reply:
Where should the 'true' ticket state live then? I don't think it should be within the useForm hook itself (as it is now in this PR). What about in future if we want to place trades from outside of the deal ticket (eg trading from the chart), or perhaps as we do now with arming from the orderbook? Now there is also no guarantee the deal ticket form will even be showing. 

Overall I think we should probably just using a fully controlled form (controlled from the zustand store), probably completely removing useForm. Not for this PR though 😛

Reply:
Again, resolving so as not to block